src/Compilers/CSharp/Portable/Emitter/Model/PEModuleBuilder.cs

```diff
@@ -1970,6 +1970,28 @@ internal MethodSymbol EnsureInlineArrayAsSpanExists(SyntaxNode syntaxNode, Named
                                                       diagnostics);
         }
 
+        internal NamedTypeSymbol EnsureInlineArrayTypeExists(SyntaxNode syntaxNode, SyntheticBoundNodeFactory factory, int arrayLength, DiagnosticBag diagnostics)
+        {
+            Debug.Assert(Compilation.Assembly.RuntimeSupportsInlineArrayTypes);
+
+            string typeName = PrivateImplementationDetails.GetInlineArrayTypeName(arrayLength);
+            var privateImplClass = GetPrivateImplClass(syntaxNode, diagnostics);
+            var typeAdapter = privateImplClass.GetType(typeName);
```

Comment:
>GetType [](http://example.com/codeflow?start=47&length=7)

nit: consider calling this `TryGetType` since it can fail to return a type. Then maybe consider using a `TryGet` API pattern (return bool, out parameter).

Reply:
I went with `GetType()` to match the existing `GetMethod()` method used in the other `Ensure` methods here.